tests/common/devices/sonic.py

```diff
@@ -904,10 +904,13 @@ def get_ip_route_info(self, dstip, ns=""):
             # parse set_src
             m = re.match(r"^(default|\S+) proto (zebra|bgp|186) src (\S+)", rt[0])
             m1 = re.match(r"^(default|\S+) via (\S+) dev (\S+) proto (zebra|bgp|186) src (\S+)", rt[0])
+            m2 = re.match(r"^(default|\S+) nhid (\d+) proto (zebra|bgp|186) src (\S+)", rt[0])
```

Comment:
wonder if you could add example like above. it seems the kernel changed the format.

Reply:
It could be worth it to also consider using the JSON output instead of trying to parse text. That might make it more resilient to formatting/output changes.